scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2878,252 @@ cdef class Rotation:
+        # other vectors to.
+        weight_inf = np.isposinf(weights)
+        if sum(weight_inf) > 1:
+            raise ValueError("Only one infinite weight is allowed")
+        elif N >= 2 and weight_inf.any():
+            if N == 2:
+                # A single non-infinite weight vector only needs normalization
+                a_non_inf = a[~weight_inf].reshape(3)
+                b_non_inf = b[~weight_inf].reshape(3)
+            else:  # N > 2
+                # Multiple non-infinite weight vectors need to be best aligned
+                R, _ = cls.align_vectors(a[~weight_inf], b[~weight_inf],
+                                         weights=weights[~weight_inf])
+                R_mat = R.as_matrix()
+                a_non_inf = R_mat[0, :]  # Already normalized
+                b_non_inf = R_mat[1, :]
+
+            # Reshape to 1D vectors
+            a_inf = a[weight_inf].reshape(3)
+            b_inf = b[weight_inf].reshape(3)
+
+            # Calculate angles between the primary and secondary vectors
+            a_angle = acos(np.dot(a_inf, a_non_inf))
+            b_angle = acos(np.dot(b_inf, b_non_inf))
+
+            # If the primary and secondary vectors are not too close to
+            # have numerical stability issues, fit orthonormal frames and
+            # find the rotation between them
+            if a_angle >= atol and b_angle >= atol:
+                A = np.array(_matrix_from_pri_sec_vectors(a[0], a[1]))
+                B = np.array(_matrix_from_pri_sec_vectors(b[0], b[1]))
+                R_opt = cls.from_matrix(A.T @ B)
+
+                # Note: we apply to the original vectors, since if N > 2 and we
+                # best fit the non-infinite weight vectors, we need to
+                # calculate rssd on the full set. The error will be 0 for the
+                # infinite weight vector.
+                weights_non_inf = weights.reshape(len(weights), 1)
+                weights_non_inf[weight_inf] = 0
+                a_est = R_opt.apply(b)
+                rssd = np.sqrt(np.sum(weights_non_inf * (a_est - a)**2))
+
+                if return_sensitivity:
+                    # Note: the sensitivity is not meaningful with an infinite
+                    # weight, so we return inf for all values
+                    sensitivity = np.ones([3, 3]) * np.inf
+                    return R_opt, rssd, sensitivity
+                else:
+                    return R_opt, rssd
+
+            # If the primary and secondary vectors are too close, we pull
+            # out the primary vectors and continue on to find the shortest
+            # path rotation between them
+            else:
+                a3 = a_inf
+                b3 = b_inf
+                N = 1
+
+        # If we have only one vector pair, we exactly align `b` to `a` using
+        # the shortest path rotation
+        if N == 1:
+            theta = acos(np.dot(a3, b3))
```

Comment:
I don't understand one thing.

The constrained algorithm with 2 vectors does something very similar on the first step (aligning primary vectors). But here it is another code branch and `_matrix_from_pri_sec_vectors` works separately and differently. 

Why?

Reply:
The first path with two vectors only computes this angle between `a[0]` and `[a1]`, and `b[0]` and `[b1]`, for the purpose of checking for parallelism. `_matrix_from_pri_sec_vectors` is then used to calculate the A and B frames to impose that secondary axis constraint.

We could use a similar code path for the 1-D case, by doing something like the below:
```
A = np.array(_matrix_from_pri_sec_vectors(a[0], [1, 0, 0]))
B = np.array(_matrix_from_pri_sec_vectors(b[0], [1, 0, 0]))
R_opt = cls.from_matrix(A.T @ B)
```

But then the `[1, 0, 0]` secondary axis is arbitrary, runs into potential issues if it is parallel to one of the other vectors, and will generally not correspond to the minimal angle rotation.

Reply:
I've outlined the 2 steps algorithm for 1 strict + n other alignment: first compute the strict rotation (as in this code path), then align other vectors (if there are any) using essentially 2D alignment (with atan2). 

Not necessary to follow this path, but considering there are apparent issues with reusing standard alignment on step 2 (discussed above), maybe it's a way to go.